### PR TITLE
Fix: Make sure to correctly resize the table

### LINKED ISSUE #5966: Sort out column resizing when it shrinks back from full screen Table visualization
### Discord username

_No response_

### How important is this feature to you?

3 – Lack of it makes using Enso slightly harder

### Describe the idea you'd like to see implemented.

When user opens full screen table visualization (double space) and then shrink back to default size (single space) the column size is not shrinking back to the default size as it should.

### Is your feature request related to a problem?

_No response_

### Screenshots, screencasts, mockups.

The video of current behavior that should be improved:

https://user-images.githubusercontent.com/12892578/225589002-ee3b4d91-e916-4866-bcd5-03f12536d83e.mov

### Would you be willing to help us implement this feature?

No

### PR DESCRIPTION
**Pull request by Nctdt**
### Pull Request Description
Fixes #5966;

https://user-images.githubusercontent.com/52877578/226659132-c6ffc40b-e495-4799-bd35-2c56b17086b0.mp4

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui/view/graph-editor/src/builtin/visualization/java_script/table.js [hunk 1]
```diff
@@ -231,9 +231,12 @@ class TableVisualization extends Visualization {
             const height = this.dom.getAttributeNS(null, 'height')
             const tblViewStyle = `width: ${width}px; height: ${height}px; overflow: scroll;`
             this.tabElem.setAttributeNS(null, 'style', tblViewStyle)
+            // Ensure that this.tabElem and this.dom have been updated before calling
+            // sizeColumnsToFit() to get the latest element styles.
```
Comment (wdanilo) [inline app/gui/view/graph-editor/src/builtin/visualization/java_script/table.js]: Could you please describe it better. If I did not know what this is about, after reading this comment in the code I would not understand it at all. Include in the description info why this is here, how it affects visualization and why it affects it so, please :)

Comment (Nctdt) [reply app/gui/view/graph-editor/src/builtin/visualization/java_script/table.js]: This is a problem caused by browser reflow. I searched for a long time but couldn't find the demo I had seen before, and I forgot how to reproduce it. However, I will try my best to explain it in the comments.

Comment (wdanilo) [reply app/gui/view/graph-editor/src/builtin/visualization/java_script/table.js]: This even more shows that this doc comment should be more clear, as a potential reader in the future should be able to not only understand it, but should know how to check it / reproduce it. If we have problem with that already, we need to improve it definitely :)

File: app/gui/view/graph-editor/src/builtin/visualization/java_script/table.js [hunk 2]
```diff
@@ -231,9 +231,23 @@ class TableVisualization extends Visualization {
...
+             * This is because the browser's reflow mechanism may prevent `sizeColumnsToFit()` from obtaining
+             * the latest width of `this.tabElem` internally,
+             * resulting in the computed column width being the same as before and having no effect.
+             * Therefore, using `requestAnimationFrame` function ensures that the latest available width is correctly obtained
+             * within `sizeColumnsToFit()` to solve this issue.
+             *
+             * But I couldn't reproduce a simple demo, even after examining the source code corresponding to agGrid.
+             * It's also possible that this is due to the interaction with wasm,
+             * so this theory may also be incorrect.
+             */
```
Comment (wdanilo) [inline app/gui/view/graph-editor/src/builtin/visualization/java_script/table.js]: 1. Formatting of the comment is broken.
2. `Therefore, using `requestAnimationFrame` function ensures that the latest available width is correctly obtained` - hmm, I still don't understand why. The docs state that this ensures that the latest width is obtained, but how does it ensures it?
3. I don't believe there is any WASM interaction here, this is plain JS visualization.

Comment (Nctdt) [reply app/gui/view/graph-editor/src/builtin/visualization/java_script/table.js]: `I don't believe there is any WASM interaction here, this is plain JS visualization.` - yeah, you're right.

**Review by jdunkerley — APPROVED** [review]

**Comment by sylwiabr** [comment]
@Nctdt is this ready to be merged?

**Comment by Nctdt** [comment]
> @Nctdt is this ready to be merged?

Yes, we can merge now.

**Pull request merged**